util/sessionutil/session.go

```diff
@@ -0,0 +1,113 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sessionutil
+
+import (
+	"context"
+	"time"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/tidb/kv"
+	"github.com/pingcap/tidb/metrics"
+	"github.com/pingcap/tidb/parser/terror"
+	"github.com/pingcap/tidb/sessionctx"
+	"github.com/pingcap/tidb/sessiontxn"
+	"github.com/pingcap/tidb/util/chunk"
+	"github.com/pingcap/tidb/util/sqlexec"
+)
+
+// Session wraps sessionctx.Context for transaction usage.
+type Session struct {
+	sessionctx.Context
+}
+
+// NewSession is return new point of session to be used.
+func NewSession(s sessionctx.Context) *Session {
+	return &Session{Context: s}
+}
+
+// Begin is to begin a new transaction.
+func (s *Session) Begin() error {
+	err := sessiontxn.NewTxn(context.Background(), s)
+	if err != nil {
+		return err
+	}
+	s.GetSessionVars().SetInTxn(true)
+	return nil
+}
+
+// Commit is commit one transaction.
+func (s *Session) Commit() error {
+	s.StmtCommit(context.Background())
+	return s.CommitTxn(context.Background())
+}
+
+// GetTxn return active transaction.
+func (s *Session) GetTxn() (kv.Transaction, error) {
+	return s.Txn(true)
+}
+
+// Rollback is rollback a transaction.
+func (s *Session) Rollback() {
+	s.StmtRollback(context.Background(), false)
+	s.RollbackTxn(context.Background())
+}
+
+// Reset is reset statement.
+func (s *Session) Reset() {
+	s.StmtRollback(context.Background(), false)
+}
+
+// Execute is used to run one query internal.
+func (s *Session) Execute(ctx context.Context, query string, label string) ([]chunk.Row, error) {
+	startTime := time.Now()
+	var err error
+	defer func() {
+		metrics.DDLJobTableDuration.WithLabelValues(label + "-" + metrics.RetLabel(err)).Observe(time.Since(startTime).Seconds())
```

Comment:
Maybe it should be a submodule of ddl, since it is always using DDL source type and metrics?

Reply:
No， it will be a common module, I will see how to deal with this.